### ISSUE TITLE
Metadata Editor: allow for setting and display of a default value for text* input

### ISSUE THREAD

**Issue by ptitzler** [elyra-ai/elyra#1324]
In certain scenarios a [text] input might implicitly be assigned a default value that is applied if no value is provided. As I understand it (correct me if I am wrong), this is how this is handled today:
- Metadata/UI: input field is marked as optional
- Metadata: contains `default` property, which defines the value that is provided to the back-end

Desired behavior:
- Metadata/UI: input field is marked as required
- Metadata contains `default` property, which defines the value that is provided to the back-end
- The input field is pre-populated with the value. The user can override/customize the value if desired.

Benefit:
- The user has a clear understanding what the default is.

Example use case:
- An Airflow Runtime Configuration contains a property "git server", which identifies the endpoint of a git server, e.g. "https://api.github.com". When the user creates a new runtime configuration this text input field is pre-populated with the default `https://api.github.com`.
- If the user clears the input an error is raised (property is required)
- If the user enters a different value, the property is set to that value.

**Comment by akchinSTC** [comment]
Example of the default property in the runtime schema:
```
 "github_api_endpoint": {
          "title": "Github API Endpoint",
          "description": "The Github Server URL / API endpoint - Public or Enterprise",
          "type": "string",
          "format": "uri",
          "default": "https://api.github.com",
          "uihints": {
            "category": "Apache Airflow"
          }
 ```

**Comment by ajbozarth** [comment]
I believe this can be addressed by making this line https://github.com/elyra-ai/elyra/blob/4c323e1a8b9dd55ef41d2c77a87992ffd9ca560b/packages/metadata-common/src/MetadataEditor.tsx#L448 a ternary that sets to default if the value is empty

**Comment by karlaspuldaro** [comment]
Closed by #1341
